lib/model/model.go

```diff
@@ -1564,12 +1764,15 @@ func (m *model) Request(deviceID protocol.DeviceID, folder, name string, size in
 	if err := readOffsetIntoBuf(folderFs, name, offset, res.data); fs.IsNotExist(err) {
 		l.Debugf("%v REQ(in) file doesn't exist: %s: %q / %q o=%d s=%d", m, deviceID, folder, name, offset, size)
 		return nil, protocol.ErrNoSuchFile
+	} else if err == io.EOF && len(hash) == 0 {
+		// Read beyond end of file when we can't verify the hash -- this is
+		// a padded read for an encrypted file. It's fine.
```

Comment:
How does this work? We lie about the block sizes in encrypted files, so clients ask to read more than what's available?

Reply:
The last block can have arbitrary size, but untrusted devices don't know that size, thus just request the standard block size.

Reply:
Why don't we just pad the last block?

Reply:
The sent data is already padded, because `res.data` is created with the requested size. Thus reading from file tries to read beyond the end of the file.